### PR TITLE
ci: Add Java Smoke Tests

### PR DESCRIPTION
**Pull request by averikitsch**

### CODE REVIEW

File: .github/workflows/test_clients.yaml [hunk 1]
```diff
@@ -29,18 +29,54 @@ jobs:
...
+      DATA_SOURCE_CHECKOUT_PATH: google-cloudevents
+    steps:
+      - name: Java Library > Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          repository: googleapis/google-cloudevents-java
+          path: ${{ env.LIBRARY_CHECKOUT_PATH }}
+          ref: snowpea-build
+
+      - name: Proto Schemas > Checkout Repository
```
Comment (grayside) [inline .github/workflows/test_clients.yaml:62]: I wonder if we could do a single checkout of proto schemas and reuse across all languages?

Comment (averikitsch) [reply .github/workflows/test_clients.yaml:62]: I don't think it's worth the work trying to create a parent job to checkout and have child jobs uses that one source. Are you concerned or just trying to minimize each language setup?

File: .github/workflows/test_clients.yaml [hunk 2]
```diff
@@ -29,18 +29,53 @@ jobs:
...
+    env:
+      LIBRARY_CHECKOUT_PATH: library
+      DATA_SOURCE_CHECKOUT_PATH: google-cloudevents
+    steps:
+      - name: Java Library > Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          repository: googleapis/google-cloudevents-java
+          path: ${{ env.LIBRARY_CHECKOUT_PATH }}
+          ref: snowpea-build
```
Comment (averikitsch) [inline .github/workflows/test_clients.yaml:60]: Note to self: remove before merging

File: .github/workflows/test_clients.yaml (hunk 1 above)
Comment (grayside) [reply .github/workflows/test_clients.yaml:62]: Just thinking about avoiding redundant actions and per-language complexity. But agreed, refactoring for that is a good future improvement.

**Review by grayside — APPROVED** [review]

**Pull request merged**